### PR TITLE
Phyplus rel 1.5

### PR DESCRIPTION
**Pull request by chrade**
## Summary 
phyplus release version 1.5 , need notice this version should revert follow commit..
fb43fd73ed9d2b60984c3e0c5dd71ea9e25d7fcd
otherwise , system startup will generate a hard fault.

### CODE REVIEW

File: boards/arm/phy62xx/phy6222/scripts/flash.ld [hunk 1]
```diff
@@ -88,8 +88,53 @@ SECTIONS
...
+        *libphy6222_rf.a:patch.o(.text.llCalcMaxScanTime1)
+        *libphy6222_rf.a:patch.o(.text.LL_SetAdvControl1)
+        *libphy6222_rf.a:patch.o(.text.llSetupSecAdvEvt1)
+        *libphy6222_rf.a:patch.o(.text.ll_scheduler2)
+        *libphy6222_rf.a:patch.o(.text.llSetupNextSlaveEvent1)
+		
+		*libapps.a:flash.c.*.o(.text .text.*)
+		*libapps.a:*.o(.text.drv_disable_irq1)
+		*libapps.a:*.o(.text.drv_enable_irq1)
+
```
Comment (LingaoM) [inline boards/arm/phy62xx/phy6222/scripts/flash.ld:132]:
```suggestion
        *libphy6222_rf.a:patch.o(.text.llSetupNextSlaveEvent1)
		
	*libapps.a:flash.c.*.o(.text .text.*)
	*libapps.a:*.o(.text.drv_disable_irq1)
	*libapps.a:*.o(.text.drv_enable_irq1)

```

File: boards/arm/phy62xx/phy6222/scripts/flash.ld [hunk 2]
```diff
@@ -121,6 +167,9 @@ SECTIONS
 		
         *libmm.a:circbuf.o(.text .text.*)
 		
+		*libc.a:lib_libvsprintf.o(.text .text.*)
+		*libc.a:lib_printf.o(.text .text.*)
+		*libc.a:lib_vfprintf.o(.text .text.*)
 		*libc.a:lib_skipspace.o(.text .text.*)
         *libc.a:lib_sprintf.o(.text .text.*)
```
Comment (LingaoM) [inline boards/arm/phy62xx/phy6222/scripts/flash.ld:174]:
```suggestion
        *libmm.a:circbuf.o(.text .text.*)
		
	*libc.a:lib_libvsprintf.o(.text .text.*)
	*libc.a:lib_printf.o(.text .text.*)
	*libc.a:lib_vfprintf.o(.text .text.*)
	*libc.a:lib_skipspace.o(.text .text.*)
        *libc.a:lib_sprintf.o(.text .text.*)
```

File: arch/arm/src/phy62xx/uart.c [hunk 3]
```diff
@@ -62,6 +68,9 @@ typedef struct _uart_Context
   uint8_t       tx_state;
   uart_Tx_Buf_t tx_buf;
   uart_Cfg_t    cfg;
+  uint8_t fifo_buf_store[UART_RX_BUFFER_SIZE];
+  int buf_head;
+  int buf_tail;
 } uart_Ctx_t;
 
 static uart_Ctx_t m_uartCtx[2] =
```
Comment (LingaoM) [inline arch/arm/src/phy62xx/uart.c:76]: Why reserve so much memory for devices with limited available resources? Is it necessary to put `UART_RX_BUFFER_SIZE` into kconfig?

File: arch/arm/src/phy62xx/phy62xx_exception.S [hunk 4]
```diff
@@ -339,6 +340,13 @@ exception_common_inline:
 
 	.size	exception_common_inline, .-exception_common_inline
 
+/****************************************************************************
+ *  Name: g_intstackalloc/g_intstacktop
+ *
+ * Description:
+ *   Shouldn't happen
+ *
+ ****************************************************************************/
```
Comment (LingaoM) [inline arch/arm/src/phy62xx/phy62xx_exception.S:349]: Already exists below,

File: arch/arm/src/phy62xx/pplus_mtd_flash.c [hunk 5]
```diff
@@ -144,7 +147,10 @@ static int pplus_fls_erase_chip(struct pplus_fls_dev_s *priv)
 
   for (i = 0; i < priv->nsectors; i++)
     {
+      _HAL_CS_ALLOC_();
+      HAL_ENTER_CRITICAL_SECTION();
       hal_flash_erase_sector(address);
+      HAL_EXIT_CRITICAL_SECTION();
```
Comment (LingaoM) [inline arch/arm/src/phy62xx/pplus_mtd_flash.c:153]: During critical time period, is it an appropriate way to erase flash?

Comment (LingaoM) [reply arch/arm/src/phy62xx/pplus_mtd_flash.c:153]: @xiaoxiang781216

File: boards/arm/phy62xx/phy6222/configs/newostest/defconfig.bak [hunk 6]
```diff
@@ -0,0 +1,147 @@
+#
```
Comment (CV-Bowen) [inline boards/arm/phy62xx/phy6222/configs/newostest/defconfig.bak:1]: do not need the defconfig.bak

File: boards/arm/phy62xx/phy6222/configs/newostest/defconfig [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+CONFIG_PREALLOC_TIMERS=0
+CONFIG_PSEUDOFS_ATTRIBUTES=y
+CONFIG_PSEUDOFS_SOFTLINKS=y
+CONFIG_PTHREAD_CLEANUP=y
+CONFIG_PTHREAD_CLEANUP_STACKSIZE=2
+CONFIG_PTHREAD_MUTEX_TYPES=y
+CONFIG_PTHREAD_SPINLOCKS=y
+CONFIG_PTHREAD_STACK_DEFAULT=3072
+CONFIG_RAM_SIZE=58368
+CONFIG_RAM_START=0x1fff1c00
```
Comment (CV-Bowen) [inline boards/arm/phy62xx/phy6222/configs/newostest/defconfig:110]: the CONFIG_RAM_START and CONFIG_RAM_SIZE should be equal to the value defined in flash.ld.
In flash.ld, the CONFIG_RAM_START = 0x1fff1880 and CONFIG_RAM_SIZE = 59264

File: boards/arm/phy62xx/phy6222/src/bringup.c [hunk 8]
```diff
@@ -125,16 +137,26 @@ int phy62xx_bringup(void)
 
       /* Mount the LittleFS file system */
 
-      ret = nx_mount("/dev/mtd", "/mnt/lfs", "littlefs", 0,
+      ret = nx_mount("/dev/mtd", "/data", "littlefs", 0,
                 "forceformat");
```
Comment (CV-Bowen) [inline boards/arm/phy62xx/phy6222/src/bringup.c:141]: the "forceformat" should be modified to "autoformat"

File: boards/arm/phy62xx/phy6222/configs/newostest/defconfig.bak (hunk 6 above)
Comment (chrade) [reply boards/arm/phy62xx/phy6222/configs/newostest/defconfig.bak:1]: ok. i will delete it.

File: boards/arm/phy62xx/phy6222/scripts/flash.ld (hunk 1 above)
Comment (chrade) [reply boards/arm/phy62xx/phy6222/scripts/flash.ld:132]: i will fit it.

File: arch/arm/src/phy62xx/uart.c (hunk 3 above)
Comment (chrade) [reply arch/arm/src/phy62xx/uart.c:76]: ok, i will reduce the RX_BUFF_SIZE and self check first.

File: boards/arm/phy62xx/phy6222/src/bringup.c (hunk 8 above)
Comment (chrade) [reply boards/arm/phy62xx/phy6222/src/bringup.c:141]: ok , i will fix it.

File: boards/arm/phy62xx/phy6222/configs/newostest/defconfig (hunk 7 above)
Comment (chrade) [reply boards/arm/phy62xx/phy6222/configs/newostest/defconfig:110]: i will check and fix it.

**Pull request closed without merging**